client/asset/dcr/dcr.go

```diff
@@ -2846,6 +2942,60 @@ func (dcr *ExchangeWallet) getBestBlock(ctx context.Context) (*block, error) {
 	return &block{hash: hash, height: height}, nil
 }
 
+func (dcr *ExchangeWallet) getBlock(ctx context.Context, blockHash *chainhash.Hash, verboseTx bool) (*chainjson.GetBlockVerboseResult, error) {
+	blockVerbose, err := dcr.wallet.GetBlockVerbose(ctx, blockHash, verboseTx)
+	if err != nil {
+		return nil, fmt.Errorf("error retrieving block %s: %w", blockHash, translateRPCCancelErr(err))
```

Comment:
The caller knows the block hash they requested, and the non-nil error tells them all they need to know.  Together with my comment about `translateRPCCancelErr` being done in the `dcr.Wallet` implementer, this will become a oneliner.